planner: update the doc about Plan Cache (#17967)
pingcap/docs: optimizer-hints.md

prepare stmt from 'select  /*+ IGNORE_PLAN_CACHE() */ * from t where t.id = ?';
```

### SET_VAR(VAR_NAME=VAR_VALUE)

You can temporarily modify the value of system variables during statement execution by using the `SET_VAR(VAR_NAME=VAR_VALUE)` hint. After the statement is executed, the value of the system variable in the current session is automatically restored to the original value. This hint can be used to modify some system variables related to the optimizer and executor. For a list of system variables that can be modified using this hint, refer to [System variables](/system-variables.md).

> **Warning:**
>
> - It is strongly recommended not to modify variables that are not explicitly supported, as this might cause unpredictable behavior.

> > - Do not write `SET_VAR` in subqueries. Otherwise it might not work. For more information, see [`SET_VAR` does not work in subqueries](#set_var-does-not-work-in-subqueries).

Reviewer: ```suggestion
> - Do not write `SET_VAR` in subqueries. Otherwise it might not take effect. For more information, see [`SET_VAR` does not take effect when written in subqueries](#set_var-does-not-take-effect-when-written-in-subqueries).
```

EXPLAIN SELECT /*+ NO_MERGE_JOIN(t1) */ * FROM t1, t2 WHERE t1.a=t2.a;
ERROR 1815 (HY000): Internal : Can't find a proper physical plan for this query
```

> ### `SET_VAR` does not work in subqueries

Reviewer: ```suggestion
### `SET_VAR` does not take effect when written in subqueries
```

> `SET_VAR` is used to set a system variable for the current statement and should not be written in a subquery. If you write it in a subquery, `SET_VAR` might not take effect due to the special handling of subqueries.

Reviewer: ```suggestion
`SET_VAR` is used to modify the value of system variables for the current statement. Do not write it in subqueries. If you write it in a subquery, `SET_VAR` might not take effect due to the special handling of subqueries.
```

In the following example, `SET_VAR` is written in the subquery, so it does not take effect.

```sql
mysql> SELECT @@MAX_EXECUTION_TIME, a FROM (SELECT /*+ SET_VAR(MAX_EXECUTION_TIME=123) */ 1 as a) t;
+----------------------+---+
| @@MAX_EXECUTION_TIME | a |
+----------------------+---+
|                    0 | 1 |
+----------------------+---+
1 row in set (0.00 sec)
```

> In the following example, SET_VAR is not written in a subquery, so it can take effect.

Author: ```suggestion
In the following example, `SET_VAR` is not written in a subquery, so it can take effect.
```
```suggestion
In the following example, SET_VAR is not written in a subquery, so it can take effect.
```

> In the following example, `SET_VAR` is not written in a subquery, so it can take effect.

Reviewer: ```suggestion
In the following example, `SET_VAR` is not written in a subquery, so it takes effect.
```
